File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int size = s.length();
	HashSet<Character> seenCharacter = new HashSet<>();
	int left =0;
	int maxLength=0;
	for(int i=0;i<size;i++){
		char ch = s.charAt(i);
		while(seenCharacter.contains(ch)){
			char windowStartCharacter = s.charAt(left);
			seenCharacter.remove(windowStartCharacter);
			left++;
		}
seenCharacter.add(ch);
		maxLength= Math.max(maxLength, seenCharacter.size());
		
	}
return maxLength;

    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        HashSet<char> MySet= new HashSet<char>();
        int start =0,max=0;
        for(int i=0; i< s.Length; i++){
            while(MySet.Contains(s[i])){
                MySet.Remove(s[start++]);
            }
            MySet.Add(s[i]);
            max= Math.Max(max, i-start+1);
        }
        return max;
    }
}